src/OSPSuite.SimModel/src/FormulaChange.cpp

```diff
@@ -59,6 +59,12 @@ void FormulaChange::XMLFinalizeInstance (const XMLNode & pNode, Simulation * sim
 
 	//mark quantity as one (potentially) changed by switch(es)
 	_quantity->SetIsChangedBySwitch(true);
+
+	//If the quantity is a species, add this formula change to the list of entities that cache its scale factor (used for updating scale factors)
+	Species * species = dynamic_cast<Species*>(_quantity);
+	if (species == NULL) //changed quantity is not a species
+		return;
+	species->AddEntityWithCachedScaleFactor(this);
 }
```

Comment:
If changed quantity is a species: add current FormulaChange to the List of entities caching species's scale factor